PR: https://github.com/odoo/odoo/pull/242503

From: 317fa1b33bf74a0a79c0f15d14ecd0a60118ac43
From: dhha-odoo
Date: 2026-06-10 17:34:15

Breaking data model changes scores: del:10 + add:0, change matches:
-    module_delivery_dhl = fields.Boolean("DHL Express Connector")
-    module_delivery_fedex_rest = fields.Boolean("FedEx Connector")
-    module_delivery_ups_rest = fields.Boolean("UPS Connector")
-    module_delivery_usps_rest = fields.Boolean("USPS Connector")
-    module_delivery_bpost = fields.Boolean("bpost Connector")
-    module_delivery_easypost = fields.Boolean("Easypost Connector")
-    module_delivery_sendcloud = fields.Boolean("Sendcloud Connector")
-    module_delivery_shiprocket = fields.Boolean("Shiprocket Connector")
-    module_delivery_starshipit = fields.Boolean("Starshipit Connector")
-    module_delivery_envia = fields.Boolean("Envia.com Connector")

Total Changes: 15

[IMP] stock: Improve UX for delivery methods and shipping connectors

Delivery methods and shipping connectors were split across different sections
in Inventory, creating a disconnected and confusing experience. The
dedicated Shipping Connectors section also took up significant
space while offering limited contextual value.

This change unifies all delivery methods into a single view and
groups all shipping connector modules in dedicated list and kanban views.
It improves clarity, reduces unnecessary navigation, groups all connectors
in one place, and streamlines the overall configuration experience.

Enterprise PR:- https://github.com/odoo/enterprise/pull/103467
Upgrade PR:- https://github.com/odoo/upgrade/pull/9208

Task - 5380595

closes odoo/odoo#242503

Related: odoo/enterprise#103467
Related: odoo/upgrade#9208
Signed-off-by: Arnold Moyaux (arm) <arm@odoo.com>

================================= pseudo patch: =================================

--- a/addons/stock/models/res_config_settings.py
+++ b/addons/stock/models/res_config_settings.py
@@ -31,16 +31,6 @@ class ResConfigSettings(models.TransientModel):
     stock_move_email_validation = fields.Boolean(related='company_id.stock_move_email_validation', readonly=False)
     module_stock_sms = fields.Boolean("SMS Confirmation")
     module_delivery = fields.Boolean("Delivery Methods")
-    module_delivery_dhl = fields.Boolean("DHL Express Connector")
-    module_delivery_fedex_rest = fields.Boolean("FedEx Connector")
-    module_delivery_ups_rest = fields.Boolean("UPS Connector")
-    module_delivery_usps_rest = fields.Boolean("USPS Connector")
-    module_delivery_bpost = fields.Boolean("bpost Connector")
-    module_delivery_easypost = fields.Boolean("Easypost Connector")
-    module_delivery_sendcloud = fields.Boolean("Sendcloud Connector")
-    module_delivery_shiprocket = fields.Boolean("Shiprocket Connector")
-    module_delivery_starshipit = fields.Boolean("Starshipit Connector")
-    module_delivery_envia = fields.Boolean("Envia.com Connector")
     module_quality_control = fields.Boolean("Quality")
     module_quality_control_worksheet = fields.Boolean("Quality Worksheet")
     group_stock_multi_locations = fields.Boolean('Storage Locations', implied_group='stock.group_stock_multi_locations',
@@ -156,3 +146,8 @@ class ResConfigSettings(models.TransientModel):
                 raise UserError(_("You have product(s) in stock that have lot/serial number tracking enabled. \nSwitch off tracking on all the products before switching off this setting."))
 
         return
+
+    def action_view_delivery_methods(self):
+        if action := self.env.ref('delivery.action_delivery_carrier_form', raise_if_not_found=False):
+            return action._get_action_dict()
+        return
